keeper-gradle-plugin/src/main/java/com/slack/keeper/KeeperPlugin.kt

```diff
@@ -246,6 +251,23 @@ public class KeeperPlugin : Plugin<Project> {
     }
   }
 
+  private fun Project.androidEmbeddedJarRegularFileProvider(
+      appExtension: AppExtension,
+      path: String,
+      required: Boolean
+  ): Provider<RegularFile> =
+    layout.file(provider {
+      val compileSdkVersion = appExtension.compileSdkVersion
+        ?: error("No compileSdkVersion found")
+      File("${appExtension.sdkDirectory}/platforms/${compileSdkVersion}/${path}").let {
+        val exists = it.exists()
+        check(required.not() or exists) {
+          "No $path found! Expected to find it at: $it"
+        }
+        it.takeIf { exists }
```

Comment:
What happens if it doesn't exist? Would gradle be unhappy with a null File provider here?

Reply:
That's a good question, so I tried to run it after moving the optional file somewhere else.
It seems Gradle is fine (I guess null means no value) despite the javadoc I found https://github.com/gradle/gradle/blob/48753466be064bbd84b643dc6995221b35084e22/subprojects/core-api/src/main/java/org/gradle/api/Project.java#L962 ...
What's sure is that R8 is failing if a --lib artifact is not found, so I decided to always provide a value, even if not existing, and I'm filtering it during the task execution.

See 0e5190a for the updated version.